### ISSUE TITLE
Problem with LDAP authorization

### ISSUE THREAD

**Issue by gwalrath** [Confluex/Zuul#86]
I have updated ldap.properties to use our AD server for authentication. That piece works fine, but I can't seem to get more than ROLE_GUEST user with my ID when I log in. I can't even create a setting - it just tells me I'm not authorized and to press the button to request access (which appears to go nowhere).

The group names in all the ldap.dn.ROLE* lines are correct, and I can see in the log that Zuul acknowledges that I am a member of those groups. The attribute for group membership, member, is correct for our AD instance.

Anyone else run in to this?

ldap.url=ldap://usdenwpvdc11.chemd.net:389
ldap.username=CN=task_iam_sync,CN=Users,DC=chemd,DC=net
ldap.password=***********
ldap.dn.ROLE_SYSTEM_ADMIN=CN=DG_IS_PRODSYSSVS,OU=GROUPS,OU=COMPANY UNIVAR USA INC,DC=CHEMD,DC=NET
ldap.dn.ROLE_ADMIN=CN=DG_IS_PRODSYSSVS,OU=GROUPS,OU=COMPANY UNIVAR USA INC,DC=CHEMD,DC=NET
ldap.dn.ROLE_USER=CN=DG_IS_PRODSYSSVS,OU=GROUPS,OU=COMPANY UNIVAR USA INC,DC=CHEMD,DC=NET
ldap.group.search.base=OU=Groups,OU=Company Univar USA Inc,DC=chemd,DC=net
ldap.group.role.attribute=distinguishedName
ldap.group.filter=member={0}
ldap.user.search.base=OU=Users,OU=Company Univar USA Inc,DC=chemd,DC=net
ldap.user.search.filter=samAccountName={0}

**Comment by psmith** [comment]
First and foremost... ldap isn't my thing.

Comparing what you have to one of mine..
my:
ldap.user.search.base=
is empty. 

Also, are you settings the active spring profile to security-ldap?

**Comment by gwalrath** [comment]
OK, I'll give that a try. Interesting.

Yes, spring is set for security-ldap - the authentication piece works fine.

**Comment by gwalrath** [comment]
Nope, that broke authentication.

**Comment by gwalrath** [comment]
Looks like I fixed it. When I updated the login to be the full DN of the user instead of the task_iam_sync@chemd.net, authentication worked fine, but not authorization. I then updated the group and user search base to be just dc=chemd,dc=net which also allowed authentication to work, which was good since I did not want to limit access to those only in this containers.

Just for kicks, I changed the search bases to be all upper-case for both groups and users: DC=CHEMD,DC=NET.

After restarting tomcat, I am now able to be authorized in the application as a system administrator.
